lib/hubspot/deprecation.rb

```diff
@@ -0,0 +1,24 @@
+module Hubspot
+  class Deprecation
+    def self.warn(function_name:, message: nil, version: "1.0")
```

Comment:
This might be overkill but I like this for two reasons:
1) I prefer writing 
`Hubspot::Deprecation.warn(function_name: :dump_properties)`
vs
`ActiveSupport::Deprecation.new("1.0", Hubspot::NAME).deprecation_warning(:dump_properties)`
2) The gem name is set by default

Reply:
@edwardloveall I discovered a reason to *not* wrap `ActiveSupport::Deprecation` to the extent that I have. By taking this approach, the deprecation warning will report the caller backtrace as `hubspot-ruby/lib/hubspot/deprecation.rb:4`, which is far less helpful than showing the location of the deprecated method. I'm going to update this class we don't lose this helpful reporting.